### PR TITLE
Add page about Scala 2 `-Xsource:3` to migration guide

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

File: _overviews/scala3-migration/tooling-scala2-xsource3.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+In addition to new warnigns, the flag enables certain benign Scala 3 syntax such as `import p.*`.
+
+This page explains the details behind the flag, an overview is shown using `scalac -Xsource:help`.
+
+## Migration vs cross-building
+
+With Scala 2.13.13 and newer, the `-Xsource:3` flag comes in two variants:
+
+  - `Xsource:3` enables warnings relevant for migrating a codebase to Scala 3.
+  - `Xsoruce:3-cross` is useful for projects that cross-build between Scala 2 and 3 for a longer period of time.
```
Comment (bishabosha) [inline _overviews/scala3-migration/tooling-scala2-xsource3.md:21]:
```suggestion
  - `Xsource:3-cross` is useful for projects that cross-build between Scala 2 and 3 for a longer period of time.
```

File: _overviews/scala3-migration/tooling-scala2-xsource3.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    For certain language constructs that trigger a warning with `-Xsoruce:3`, the behavior changes to match Scala 3.
```
Comment (bishabosha) [inline _overviews/scala3-migration/tooling-scala2-xsource3.md:22]:
```suggestion
    For certain language constructs that trigger a warning with `-Xsource:3`, the behavior changes to match Scala 3.
```

**Review by bishabosha — APPROVED** [review]
I notice the 2.13.13 mention here, should I wait until that is released officially?

**Comment by SethTisue** [comment]
Yeah I'll do the final merge here.

File: _overviews/scala3-migration/tooling-scala2-xsource3.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+type: chapter
+description: This section describes the Scala 2 compiler's -Xsource:3 flag
+num: 7
+previous-page: tooling-tour
+next-page: tooling-migration-mode
+---
+
+The Scala 2.13 compiler issues helpful migration warnings with the `-Xsource:3` flag.
+Before moving to the Scala 3 compiler, it's recommended to enable this flag in Scala 2 and address the new warnings.
+In addition to new warnigns, the flag enables certain benign Scala 3 syntax such as `import p.*`.
```
Comment (SethTisue) [inline _overviews/scala3-migration/tooling-scala2-xsource3.md:12]:
```suggestion
In addition to new warnings, the flag enables certain benign Scala 3 syntaxes such as `import p.*`.
```

File: _overviews/scala3-migration/tooling-scala2-xsource3.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+This page explains the details behind the flag, an overview is shown using `scalac -Xsource:help`.
+
+## Migration vs cross-building
+
+With Scala 2.13.13 and newer, the `-Xsource:3` flag comes in two variants:
+
+  - `Xsource:3` enables warnings relevant for migrating a codebase to Scala 3.
```
Comment (SethTisue) [inline _overviews/scala3-migration/tooling-scala2-xsource3.md:20]: Maybe add something like:

> It also enables a few benign Scala 3 syntaxes such as `import foo.*`.

File: _overviews/scala3-migration/tooling-scala2-xsource3.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+## Fatal warnings and quick fixes
+
+By default, Scala 3 migration warnings emitted by Scala 2.13 are fatal, i.e., they are reported as errors.
+This can be changed using `-Wconf`, for example `-Wconf:cat=scala3-migration:w` changes them to be reported as warnings.
+Alternatively, `-Xmigration` has the same effect.
+
+The [`@nowarn` annotation](https://www.scala-lang.org/api/current/scala/annotation/nowarn.html) can be used to suppress individual warnings, which also works with fatal warnings enabled.
+
+The Scala 2.13 compiler implements quick fixes for many Scala 3 migration warnings.
+Quick fixes are displayed in Metals-based IDEs (not yet in IntelliJ), and they can be applied directly to the source code using the `-quickfix` flag, for example `-quickfix:cat=scala3-migration`.
```
Comment (SethTisue) [inline _overviews/scala3-migration/tooling-scala2-xsource3.md:35]: Not yet in IntelliJ? Really? Is there a ticket somewhere tracking that?

**Review by SethTisue — APPROVED** [review]

**Comment by SethTisue** [comment]
new page is online: https://docs.scala-lang.org/scala3/guides/migration/tooling-scala2-xsource3.html

I've linked to it from the 2.13.13 release notes and posted or commented about it on both our forums and on Reddit

**Comment by lrytz** [comment]
I wonder if we should advertise this right now, given https://github.com/scala/bug/issues/12961. Moving to `-Xsource:3-cross` changes the encoding of every synthetic case class companion, this can cause issues.

I'll update the page. In any case, the page is entirely missing the fact that case companions are affected by `3-cross`, not sure why I missed that when I wrote it.

File: _overviews/scala3-migration/tooling-scala2-xsource3.md (hunk 5 above)
Comment (lrytz) [reply _overviews/scala3-migration/tooling-scala2-xsource3.md:38]: Not sure, if we find one we should add it to https://github.com/lampepfl/dotty/issues/14904

**Comment by SethTisue** [comment]
> I'll update the page.

👍 

I figured it was okay to publish and link to, because in the short run 99% of page visitors will get there from the 2.13.13 release notes, which already highlight 12961. But yes, of course it would be good if you could update it reasonably promptly.

**Pull request merged**